src/main/java/org/apache/maven/shared/dependency/analyzer/asm/ResultCollector.java

```diff
@@ -57,15 +57,24 @@ public void addName( String name )
         }
 
         // decode arrays
-        if ( name.startsWith( "[L" ) && name.endsWith( ";" ) )
+        if ( name.charAt( 0 ) == '[' )
         {
-            name = name.substring( 2, name.length() - 1 );
+            int i = 0;
+            do
+            {
+                ++i;
+            }
+            while ( name.charAt( i ) == '[' ); // could have array of array ...
```

Comment:
can `name` contains multiple `[` ... ?
we don't check length of name - can couse `IndexOutOfBoundsException`

Reply:
name can contain any number of '[' - see [Field Descriptors](https://docs.oracle.com/javase/specs/jvms/se17/html/jvms-4.html#jvms-4.3.2) in jvm spec.
Yes, we could have IndexOutOfBoundsException if class is malformed.  Does it matter what sort of exception is thrown?

Reply:
@slawekjaranowski - any thoughts?